### PR TITLE
Define event_diff() as a separate internal function

### PR DESCRIPTION
**Pull request by jdblischak**
Motivation: https://github.com/Merck/gsDesign2/issues/235#issuecomment-1884321766

### CODE REVIEW

File: R/expected_time.R [hunk 1]
```diff
@@ -120,3 +112,24 @@ expected_time <- function(
     return(ans)
   }
 }
+
+#' Find the difference between `ahr()` and different values of `total_duration`
```
Comment (LittleBeannie) [inline R/expected_time.R:116]: I personally suggest rewording it as follows: 'Considering the enrollment rate, failure rate, and randomization ratio, calculate the difference between the targeted number of events and the accumulated events at time `x`.' Please review and let me know if you agree with this rephrased version.

**Review by LittleBeannie — COMMENTED** [review]
Hi @jdblischak , thanks for the updates! I only have 1 minor comment below.

File: R/expected_time.R (hunk 1 above)
Comment (jdblischak) [reply R/expected_time.R]: I agree that's a great title! I just pushed the change

**Review by LittleBeannie — APPROVED** [review]

**Review by nanxstats — APPROVED** [review]
Awesome! I appreciate this refactor that makes `expected_time()` more robust.
